haystack/nodes/reader/table.py

```diff
@@ -32,6 +32,70 @@
 logger = logging.getLogger(__name__)
 
 
+class BaseTableProcessor:
+    def preprocess(self, docs: List[Document]):
+        return docs
+
+    def postprocess(self, answers: List[Answer], docs: List[Document]):
+        return answers
+
+
+class TableProcessor(BaseTableProcessor):
+    def __init__(self, num_rows=7, num_cols=7, num_row_fixed=0, num_col_fixed=0) -> None:
+        self.num_rows = num_rows
+        self.num_cols = num_cols
+        self.num_row_fixed = num_row_fixed
+        self.num_col_fixed = num_col_fixed
+
+    def preprocess(self, docs: List[Document]):
+        split_docs = []
+        for doc in docs:
+            table = doc.content
+            n_row, n_col = table.shape
+            for r in range(0, n_row, self.num_rows):
+                for c in range(0, n_col, self.num_cols):
+                    table_subset = table.iloc[r : r + self.num_rows, c : c + self.num_cols].reset_index(drop=True)
+                    split_docs.append(
+                        Document(
+                            content=table_subset,
+                            content_type="table",
+                            id=doc.id,
+                            meta={"row_offset": r, "col_offset": c},
+                        )
+                    )
+        return split_docs
+
+    def _remap_offset(self, answer: Answer, doc: Document):
+        # Return offset in split table document to the original parent document space
+        row_offset = doc.meta.get("row_offset", 0)
+        col_offset = doc.meta.get("col_offset", 0)
+
+        new_offsets = []
+
+        for offset in answer.offsets_in_document:
+            # TODO: Support Span in addition to TableCell; also, look at offsets_in_context
+            new_offsets.append(TableCell(row=row_offset + offset.row, col=col_offset + offset.col))
+
+        answer.offsets_in_document = new_offsets
+        return answer
+
+    def postprocess(self, answers: List[Answer], docs: List[Document]):
+        """Keep one answer per original document and adjust the offset of the answer."""
+        doc_answers = {}
+        for answer, doc in zip(answers, docs):
+            if answer is None:
+                continue
+            # Set the answer as the document id's cannonical answer
+            # if there isn't already an answer
+            # or this answer's score is greater than the current answer for the doc
+            if doc.id not in doc_answers or doc_answers[doc.id].score < answer.score:
+                answer = self._remap_offset(answer, doc)
+                answer.document_ids = [doc.id]
+                doc_answers[doc.id] = answer
+
+        return doc_answers.values()
```

Comment:
Small technical point I think we should make this `list(doc_answers.values())` so we return a list instead of a `dict_values` object. Not a big deal since we immediately pass it to `sorted` which then converts it to a list.